tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp

```diff
@@ -0,0 +1,595 @@
+    CHECK_ITERABLE_APPROX(
+        map_one.jacobian(source_point, time, functions_of_time),
+        map_two.jacobian(source_point, time, functions_of_time));
+    CHECK_ITERABLE_APPROX(
+        map_one.inv_jacobian(source_point, time, functions_of_time),
+        map_two.inv_jacobian(source_point, time, functions_of_time));
+  };
+  const auto map2 = serialize_and_deserialize(map);
+  check_map_equality(
+      domain::make_coordinate_map<Frame::Logical, Frame::Grid>(map),
+      domain::make_coordinate_map<Frame::Logical, Frame::Grid>(map2));
+  test_helper(map2);
+}
+
+template <bool InteriorMap>
+void test_map(gsl::not_null<std::mt19937*> generator) noexcept {
+  INFO("Map");
+  std::uniform_real_distribution<> phi_dis(0, 2.0 * M_PI);
+  std::uniform_real_distribution<> theta_dis(0, M_PI);
+
+  // Create a map, choose a time, and create a FunctionOfTime.
+  // Set up the map
+  CoordinateMaps::TimeDependent::SphericalCompression<InteriorMap> map{};
+  double time{std::numeric_limits<double>::signaling_NaN()};
+  std::unordered_map<std::string,
+                     std::unique_ptr<::domain::FunctionsOfTime::FunctionOfTime>>
+      functions_of_time{};
+
+  double min_radius{std::numeric_limits<double>::signaling_NaN()};
+  double max_radius{std::numeric_limits<double>::signaling_NaN()};
+  std::array<double, 3> center{std::numeric_limits<double>::signaling_NaN()};
+  generate_map_time_and_f_of_time(
+      make_not_null(&map), make_not_null(&time),
+      make_not_null(&functions_of_time), make_not_null(&min_radius),
+      make_not_null(&max_radius), make_not_null(&center), generator);
+
+  const double theta{theta_dis(*generator)};
+  const double phi{phi_dis(*generator)};
+
+  // Choose a point of unit radius with the randomly selected theta, phi.
+  // This test will rescale this point to place it in different regions.
+  const double rho_x0{sin(theta) * cos(phi)};
+  const double rho_y0{sin(theta) * sin(phi)};
+  const double rho_z0{cos(theta)};
+
+  // A helper that returns a point given a radius
+  auto point = [&rho_x0, &rho_y0, &rho_z0, &center](const double& radius) {
+    return std::array<double, 3>{{radius * rho_x0 + center[0],
+                                  radius * rho_y0 + center[1],
+                                  radius * rho_z0 + center[2]}};
+  };
+
+  // A helper that checks if two points are scaled by the same factor
+  // or not
+  auto check_point_scale_factors =
+      [&center, &map, &time, &functions_of_time](
+          const std::array<double, 3>& orig_point_1,
+          const std::array<double, 3>& orig_point_2,
+          const bool check_if_equal) {
+        const std::array<double, 3> mapped_point_1{
+            map(orig_point_1, time, functions_of_time)};
+        const std::array<double, 3> mapped_point_2{
+            map(orig_point_2, time, functions_of_time)};
+        const std::array<double, 3> scale_factor_1{
+            {(orig_point_1[0] - mapped_point_1[0]) /
```

Comment:
this test feels like it could be nicer to read as a loop rather than explicit indices 0-2 in as many places as it is.